dsp-project/handler/src/test/scala/dsp/project/handler/ProjectHandlerSpec.scala

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.project.handler
+
+import zio._
+import zio.prelude._
+import zio.test.Assertion._
+import zio.test._
+
+import dsp.errors.DuplicateValueException
+import dsp.errors.NotFoundException
+import dsp.project.domain.Project
+import dsp.project.repo.impl.ProjectRepoMock
+import dsp.valueobjects.Project._
+import dsp.valueobjects.ProjectId
+import dsp.valueobjects.V2
+
+object ProjectHandlerSpec extends ZIOSpecDefault {
+
+  private def initializeHandler(projects: Project*): URIO[ProjectHandler with ProjectRepoMock, ProjectHandler] =
+    for {
+      repo    <- ZIO.service[ProjectRepoMock]
+      _       <- repo.initializeRepo(projects: _*)
+      handler <- ZIO.service[ProjectHandler]
+    } yield handler
+
+  private def getValidated[NonEmptyChunk[E], A](validation: Validation[Throwable, A]): A =
+    validation.fold(e => throw e.head, v => v)
+
+  private val shortCode  = getValidated(ShortCode.make("0000"))
+  private val shortCode2 = getValidated(ShortCode.make("0001"))
+  private val id         = getValidated(ProjectId.make(shortCode))
+  private val id2        = getValidated(ProjectId.make(shortCode2))
+  private val name       = getValidated(Name.make("projectName"))
+  private val name2      = getValidated(Name.make("projectName 2"))
+  private val description = getValidated(
+    ProjectDescription.make(Seq(V2.StringLiteralV2("project description", Some("en"))))
+  )
+  private val description2 = getValidated(
+    ProjectDescription.make(Seq(V2.StringLiteralV2("different project description", Some("en"))))
+  )
+  private val project  = getValidated(Project.make(id, name, description))
+  private val project2 = getValidated(Project.make(id2, name2, description2))
+  // private val project = getValidated(Project.make(id, name, description))
+  def spec = suite("ProjectHandlerSpec")(
+    getAllProjectsTests,
+    getProjectTests,
+    createProjectTests,
+    deleteProjectTests,
+    updateProjectTests
+  ).provide(ProjectRepoMock.layer, ProjectHandler.layer)
+
+  private val getAllProjectsTests = suite("get all projects")(
+    test("return an empty list when requesting all users when there are none") {
+      for {
+        handler           <- initializeHandler()
```

Comment:
so this is actually initializeRepo?

Reply:
not, it is not. in any case, a bit confusing.

Reply:
I agree that it's a bit confusing... but it's the "cleanest" solution I could come up with in terms of usage. But I'm open for suggestions how it could be done better :)  
(I did look at how ZIO does it with the testConsole etc., but they just have it defined in the ZIOSpecDefault, so in order to hide this even more, we'd need to create a subclass of ZIOSpecDefault, which wouldn't be too nice, I think)